docs/user-guide/02-convenience-features.md

```diff
@@ -0,0 +1,227 @@
+# Basic Convenience Features
+
+This topic covers a series of nice-to-have language features in Slang. These features are not supported by HLSL but are introduced to Slang to simplify code development. Many of these features are added to Slang per request of our users. 
```

Comment:
One thing that is missing from this section that might be nice is the `let` and `var` syntax for declaring variables. These cover 99% of the cases where people tend to reach for `auto`, so they seem nice to discuss.

(Also, using `let` automatically makes variables immutable by default, which is nice for those users who currently add `const` to every single variable they declare)